extensions/cms/src/cmsUtils.ts

```diff
@@ -24,9 +31,36 @@ const mssqlProvider: string = 'MSSQL';
  */
 export class CmsUtils {
 
+	private _credentialProvider: azdata.CredentialProvider;
 	private _cmsService: mssql.CmsService;
 	private _registeredCmsServers: ICmsResourceNodeInfo[];
 
+	constructor() {
+		// 1) Get a credential provider
+		// 2) Store the credential provider for use later
+		this.setupCredentialProvider();
+	}
+
+	private async setupCredentialProvider(): Promise<boolean> {
+		let credProvider = await azdata.credentials.getProvider(CredentialNamespace);
+		if (credProvider) {
```

Comment:
This is currently duplicating the logic in credentialProvider(). This could be problematic if someone creates this and thus kicks off this call but then something else like savePassword is called before this finishes and thus credentialProvider() starts fetching it as well, which would result in not only unnecessary duplication of the call but also possible race condition issues where the same provider isn't used for different calls.

As I suggested if you're going to go this method vs the separate setup call then you would suggest storing the Promise returned by this method and then awaiting on that Promise in each of the calls below. So it'd look like

await _fetchCredProvider: Promise<void>;
this._credentialProvider.... // This will be fetched at this point so you can use it as normal